### PR TITLE
Finalize ProductionERC20Token

### LINKED ISSUE #426: Allow admin increase of ERC20Position size [1]

### PR DESCRIPTION
**Pull request by BrendanChou**
Finalize the modified Margin Tokens that will actually be used in production

fixes #426 
fixes part of #430

### CODE REVIEW

File: contracts/margin/external/ERC20/ERC20CappedLong.sol [hunk 1]
```diff
@@ -51,7 +66,18 @@ contract ERC20CappedLong is ERC20Long {
             trustedWithdrawers
         )
     {
-        tokenCap = cap;
+        setTokenCapInternal(cap);
+    }
+
+    // ============ Public Functions ============
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20CappedLong.sol]: call this Owner Functions or similar

File: contracts/margin/external/ERC20/ERC20CappedLong.sol [hunk 2]
```diff
@@ -72,4 +98,15 @@ contract ERC20CappedLong is ERC20Long {
 
         return tokenAmount;
     }
+
+    function setTokenCapInternal(
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ERC20CappedLong.sol]: Is there a reason you have this as a private function and don't just put those two lines in `setTokenCap`?

Comment (BrendanChou) [reply contracts/margin/external/ERC20/ERC20CappedLong.sol]: It's in the constructor too

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import { MarginHelper } from "../lib/MarginHelper.sol";
+
+
+/**
+ * @title ProductionERC20Long
+ * @author dYdX
+ *
+ * Early production version of an ERC20Long with a token cap and a trusted closer
+ */
+contract ProductionERC20Long is
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: `ProductionERC20Long` is a bad name. It makes it sound like all the other ones are just for dev. I think just `CappedERC20Long` is fine

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    // ============ Events ============
+
+    event TokenCapSet(
+        uint256 tokenCap
+    );
+
+    // ============ State Variables ============
+
+    address public TRUSTED_LATE_CLOSER;
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: make this a mapping of addresses

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    // ============ Constructor ============
+
+    constructor(
+        bytes32 positionId,
+        address margin,
+        address initialTokenHolder,
+        address[] trustedRecipients,
+        address[] trustedWithdrawers,
+        uint256 cap,
+        address trustedLateCloser
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: make this an array of addresses

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+        require(
+            totalSupply_.add(tokenAmount) <= tokenCap,
+            "ProductionERC20Long#getTokenAmountOnAdd: Adding tokenAmount would exceed cap"
+        );
+
+        return tokenAmount;
+    }
+
+    function closeUsingTrustedRecipient(
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: add a comment that this is overriding `ERC20Position`'s `closeUsingTrustedRecipient`. It confused me for a moment

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            trustedRecipients,
+            trustedWithdrawers
+        )
+    {
+        TRUSTED_LATE_CLOSER = trustedLateCloser;
+        tokenCap = cap;
+    }
+
+    // ============ Owner-Only Functions ============
+
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: The owner should also be able to add and remove trusted late closers. We may need to do this if we update the address we're using to send transactions in our services, etc

File: contracts/margin/external/ERC20/ProductionERC20Long.sol [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        address closer,
+        address payoutRecipient,
+        uint256 requestedAmount
+    )
+        internal
+        returns (uint256)
+    {
+        MarginCommon.Position memory position = MarginHelper.getPosition(DYDX_MARGIN, POSITION_ID);
+
+        bool afterEnd =
```
Comment (AntonioJuliano) [inline contracts/margin/external/ERC20/ProductionERC20Long.sol]: [General comment] I don't like the architecture you're using here. You have to copy a ton of stuff between this and the short version. We should at least share `closeUsingTrustedRecipient` and the state variables

File: contracts/margin/external/payoutrecipients/DutchAuctionCloser.sol [hunk 9]
```diff
@@ -187,12 +197,12 @@ contract DutchAuctionCloser is
...
     }
 
-    // ============ Private Helper-Functions ============
+    // ============ Internal Helper-Functions ============
 
     function getAuctionTimeLimits(
         bytes32 positionId
     )
-        private
+        internal
```
Comment (AntonioJuliano) [inline contracts/margin/external/payoutrecipients/DutchAuctionCloser.sol]: why was this changed?
